File: starter-bots/consoleClient/NETCoreBot/Services/BotService.cs
using NETCoreBot.Models;
using System;

namespace NETCoreBot.Services
{
    public class BotService
    {
        private GameObject _bot;
        private PlayerAction _playerAction;
        private GameState _gameState;

        public BotService()
        {
            _bot = new GameObject();
            _playerAction = new PlayerAction();
            _gameState = new GameState();
        }

        public GameObject GetBot()
        {
            return _bot;
        }

        public PlayerAction GetPlayerAction()
        {
            return _playerAction;
        }

        public void SetBot(GameObject bot)
        {
            _bot = bot;
        }

        public void SetPlayerAction(PlayerAction playerAction)
        {
            var random = new Random();
            var actionId = random.Next(1, 10);
            var heading = random.Next(0, 360);

            playerAction.ActionId = actionId;
            playerAction.Heading = heading;

            _playerAction = playerAction;
        }

        public GameState GetGameState()
        {
            return _gameState;
        }

        public void SetGameState(GameState gameState)
        {
            _gameState = gameState;
        }
    }
}

File: starter-bots/PythonBot/BotService.py
from PlayerAction import PlayerAction
from PlayerActions import PlayerActions
import random


class BotService:
    def __init__(self, bot=None, playerAction=None, gameState=None):
        self.bot = bot
        self.playerAction = playerAction
        self.gameState = gameState

    def computeNextPlayerAction(self, id):
        self.playerAction = {"PlayerId": id, "Action": PlayerActions.Forward.value, "Heading": random.randint(0, 359)}
        return self.playerAction

    def set_bot(self, args):
        self.bot = args

    def set_game_state(self, args):
        self.gameState = args
        print("Receive GameState")

    def set_playerAction_id(self, id):
        self.playerAction.PlayerId = id
